gulpfile.js

```diff
@@ -67,24 +66,28 @@ gulp.task('jshint', function () {
         .pipe(jshint.reporter('fail'));
 });
 
-gulp.task('jshint:watch', function () {
-    gulp.watch(PROJECT_PATTERNS.lint, ['jshint']);
-});
-
-gulp.task('js', ['jshint', 'jscs']);
-
-gulp.task('js:watch', function () {
-    gulp.watch(PROJECT_PATTERNS.lint, ['js']);
-});
-
+// #############################################################################
+// JS UNIT-TESTS
 gulp.task('tests:unit', function (done) {
     new Server({
         configFile: PROJECT_PATH.tests + '/karma.conf.js',
         singleRun: true
     }, done).start();
 });
+gulp.task('tests:watch', function () {
+    gulp.watch(PROJECT_PATTERNS.lint, ['tests:unit']);
+});
 
+// #############################################################################
+// TASKS
+gulp.task('js', gulpsync.sync(['jshint', 'jscs', 'tests:unit']));
```

Comment:
Yes, it slightly improves output on ci and during local tests and doesn't affect the speed a lot.